salt/modules/transactional_update.py

```diff
@@ -985,13 +985,40 @@ def call(function, *args, **kwargs):
             else:
                 return local
         except ValueError:
-            return {"result": False, "retcode": 1, "comment": ret_stdout}
+            local = {"result": False, "retcode": 1, "comment": ret_stdout}
+            return local
     finally:
         # Check if reboot is needed
-        if activate_transaction and pending_transaction():
+        if (activate_transaction and pending_transaction()) or (
+            not kwargs.get("test", False) and _user_specified_reboot(local, function)
+        ):
             reboot()
 
 
+def _user_specified_reboot(local, function):
+    if function != "state.highstate" and function != "state.sls":
```

Comment:
Looks like we need to cover all of these functions: https://github.com/openSUSE/salt/blob/86d6235df59936c038df608030a715901897b85b/salt/executors/transactional_update.py#L14-L19
Actually `state.highstate` is not widely used in Uyuni/SUMA, but mostly `state.apply` with no `mods` specified (as `state.apply` with no mods is acting as `state.highstate`)

Reply:
Indeed, great catch - didn't want to maintain the list of functions in two places, so I pulled in the `DELEGATION_MAP` directly. WTYD?